### ISSUE TITLE
[FE4] Deirdre dies for real when she is killed

### ISSUE THREAD

**Issue by Hozu** [lushen124/Universal-FE-Randomizer#372]
I believe it has to do with a value she has set in her skills 1 byte in the vanilla game.  From what I can gather, the first skill byte works as follows:

Bit 1 - Wrath
Bit 2 - Pursuit
Bit 3 - Adept
Bit 4 - ???
Bit 5 - Charm
Bit 6 - ??? - set for Deirdre and Julia
Bit 7 - Nihil
Bit 8 - Miracle

In the vanilla game, both Deirdre and Julia have the 6th bit set in the first skills byte. Similarly, Quan, Ethlyn, and Finn have the 3rd bit set in the third skills byte, when the first is for Renewal, second for Paragon, and fifth for Bargain. I believe the 3rd bit in the 3rd skills byte has to do with their different way of not dying for real. Looking through the skills of all of the playable characters, those 5 are the only ones to use those two bits in the two skills bytes. However, these bits are not set for the five characters mentioned, and when Deirdre dies, she's gone for good. She doesn't come back at the end of the chapter like she's supposed to. If I were to guess, bit 6 of skill 1 makes the character return at the end of the chapter (except endgame), while bit 3 of skill 3 makes them not count as dead when they die (for the purposes of inheritance for Quan and Ethlyn, and for Finn probably something to do with his gen 2 reappearance).

In summary, Deirdre and Julia should probably always have the 6th bit of the first skills byte set, and Quan, Ethlyn, and Finn should always have the 3rd bit in the third skill byte set, so events involving them don't break.

**Comment by Hozu** [comment]
After further testing, it turns out setting bit 6 of skill 1 enables *any* character to return from the dead at the start of the next chapter. Bootleg casual mode? However, the skill has to be already set when the unit dies, setting it after the unit dies in an attempt to bring the character back for next chapter will not work. Which only really matters if you're using cheats/editing the ROM to fix a tactical error (or just plain unfortunate randomization, looking at you Brigid).

**Comment by lushen124** [comment]
This makes a lot of sense actually. I heard there was another report about Julia not becoming "uncaptured" at the end of the chapter in Gen 2 if she is defeated.

Considering all three of Quan, Ethlyn, and Finn have the third bit set in skill 3, it probably does have to do with them all leaving if one of them dies. Finn's inheritance plays out like normal if he's paired, I believe. Combining that and the fact that Leif doesn't have any known special case like Seliph in terms of inheritance, it's unlikely this bit has something to do with how Quan and Ethlyn pass things to their children.

**Comment by Hozu** [comment]
Actually, Finn stays should Quan and Ethlyn leave if one of them die (and conversely, Quan and Ethlyn remain should Finn die). I'm guessing that bit has something to do with allowing for inheritance for their kids if/when they die? Or maybe it's a thing so they can show up later even if they die. Something like that. I haven't allowed them to die so I don't know exactly what would happen. But it's for the best to ensure those skill bits are set.

**Comment by lushen124** [comment]
Agreed. That's weird though, the fact that all three of them have it. I feel like Quan and Ethlyn's inheritance behavior is too different from Finn's for them to all be lumped into the same flag. Based on what I know from FE9, it could also be a temporary kill flag for ch. 4 and 5 (and subsequently prevents revival via Valkyria). Gen 2 Finn is a different character as far as the game is concerned, so it woudn't surprise me if they just "killed" Quan, Ethlyn, and Finn to make sure they wouldn't show up in the home castle from Ch. 4 on and had to make sure they couldn't be "revived".

**Comment by Hozu** [comment]
Something I noticed, in the win/loss count in the credits, Quan and Ethlyn show up as dead in the randomizer (names greyed out), even if they survived while under player control, and died while as green units. This not the case in the vanilla game. I guess that's one of the things that flag does.

**Comment by Vennobennu** [comment]
I've had Gen 2 Finn not appear after getting Gen 1 Finn killed - I imagine the flag indicates that this unit is not to be counted as dead even when slain during gameplay.
